Framework/Algorithms/test/RebinTest.h

```diff
@@ -146,7 +199,139 @@ class RebinTest : public CxxTest::TestSuite {
+    Rebin rebin_again;
+    rebin_again.initialize();
+    rebin_again.setPropertyValue("InputWorkspace", "test_in1D");
+    rebin_again.setPropertyValue("OutputWorkspace", "test_out_again");
+    rebin_again.setPropertyValue("Params", "1.0,1.0,1000.0");
+    rebin_again.setPropertyValue("Power", "0.5");
+    rebin_again.setPropertyValue("BinningMode", "Logarithmic");
+    TS_ASSERT_THROWS_NOTHING(rebin_again.execute());
+    TS_ASSERT(rebin_again.isExecuted());
+
+    // ensure that properties were set consistent to log binning
+    TS_ASSERT(double(rebin_again.getProperty("Power")) == 0.0);
+    std::vector<double> rbParams = rebin_again.getProperty("Params");
+    TS_ASSERT(rbParams[0] == 1.0);
+    TS_ASSERT(rbParams[1] == -1.0);
+    TS_ASSERT(rbParams[2] == 1000.0);
+
+    MatrixWorkspace_sptr rebindata_again =
+        AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>("test_out_again");
+    auto &outX_again = rebindata_again->x(0);
+
+    TS_ASSERT_EQUALS(outX_default.size(), outX_again.size());
+    for (size_t i = 0; i < outX_default.size(); i++) {
+      TS_ASSERT_EQUALS(outX_default[i], outX_again[i]);
+    }
+    TS_ASSERT(checkBinWidthMonotonic(rebindata_again, Monotonic::INCREASE));
+
+    AnalysisDataService::Instance().remove("test_in1D");
+    AnalysisDataService::Instance().remove("test_out_default");
+    AnalysisDataService::Instance().remove("test_out_again");
+  }
+
+  // Check that reverse logarithmic binning is performed if the BinningMode is set,
+  // even if positive binwidth and even if UseReverseLogarithmic set to false.
+  // Run once in the default manner, then run again with BinningMode set, and compare.
+  void testworkspace1D_reverse_logarithmic_binning_modeset() {
+    Workspace2D_sptr test_in1D = Create1DWorkspace(50);
+    test_in1D->setDistribution(true);
+    AnalysisDataService::Instance().add("test_in1D", test_in1D);
+
+    Rebin rebin_default;
+    rebin_default.initialize();
+    rebin_default.setPropertyValue("InputWorkspace", "test_in1D");
+    rebin_default.setPropertyValue("OutputWorkspace", "test_out_default");
+    rebin_default.setProperty("Params", "1.0,-1.0,1000.0");
+    rebin_default.setProperty("UseReverseLogarithmic", true);
+    TS_ASSERT_THROWS_NOTHING(rebin_default.execute());
+    TS_ASSERT(rebin_default.isExecuted());
+    MatrixWorkspace_sptr rebindata_default =
+        AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>("test_out_default");
+    auto &outX_default = rebindata_default->x(0);
+
+    // now run again but with positive step, using logarithmic mode
+    Rebin rebin_again;
+    rebin_again.initialize();
+    rebin_again.setPropertyValue("InputWorkspace", "test_in1D");
+    rebin_again.setPropertyValue("OutputWorkspace", "test_out_again");
+    rebin_again.setProperty("Params", "1.0,1.0,1000.0");
+    rebin_again.setProperty("UseReverseLogarithmic", false);
+    rebin_again.setProperty("Power", "0.5");
```

Comment:
Is the power here used at all?

Reply:
It is not used by the algorithm.  It is being set to verify that if the binning mode is reverse logarithmic, the algo will not be confused by nor use the power property.